api/src/mailbox.rs

```diff
@@ -3148,6 +3173,151 @@ impl Request for CmMldsaVerifyReq {
+    pub hdr: MailboxReqHeader,
+    pub cmk: Cmk,
+}
+
+impl Request for CmEcdsaPublicKeyReq {
+    const ID: CommandId = CommandId::CM_ECDSA_PUBLIC_KEY;
+    type Resp = CmEcdsaPublicKeyResp;
+}
+
+#[repr(C)]
+#[derive(Debug, IntoBytes, FromBytes, KnownLayout, Immutable, PartialEq, Eq)]
+pub struct CmEcdsaPublicKeyResp {
+    pub hdr: MailboxRespHeader,
+    pub public_key_x: [u8; ECC384_SCALAR_BYTE_SIZE],
+    pub public_key_y: [u8; ECC384_SCALAR_BYTE_SIZE],
+}
+
+impl Default for CmEcdsaPublicKeyResp {
+    fn default() -> Self {
+        Self {
+            hdr: MailboxRespHeader::default(),
+            public_key_x: [0u8; ECC384_SCALAR_BYTE_SIZE],
+            public_key_y: [0u8; ECC384_SCALAR_BYTE_SIZE],
+        }
+    }
+}
+
+impl Response for CmEcdsaPublicKeyResp {}
+
+// CM_ECDSA_SIGN
+#[repr(C)]
+#[derive(Debug, IntoBytes, FromBytes, KnownLayout, Immutable, PartialEq, Eq)]
+pub struct CmEcdsaSignReq {
+    pub hdr: MailboxReqHeader,
+    pub cmk: Cmk,
+    pub message_size: u32,
+    pub message: [u8; MAX_CMB_DATA_SIZE],
+}
+
+impl Default for CmEcdsaSignReq {
+    fn default() -> Self {
+        Self {
+            hdr: MailboxReqHeader::default(),
+            cmk: Cmk::default(),
+            message_size: 0,
+            message: [0u8; MAX_CMB_DATA_SIZE],
+        }
+    }
+}
+
+impl CmEcdsaSignReq {
+    pub fn as_bytes_partial(&self) -> CaliptraResult<&[u8]> {
+        if self.message_size as usize > MAX_CMB_DATA_SIZE {
+            return Err(CaliptraError::RUNTIME_MAILBOX_API_REQUEST_DATA_LEN_TOO_LARGE);
+        }
+        let unused_byte_count = MAX_CMB_DATA_SIZE - self.message_size as usize;
+        Ok(&self.as_bytes()[..size_of::<Self>() - unused_byte_count])
+    }
+
+    pub fn as_bytes_partial_mut(&mut self) -> CaliptraResult<&mut [u8]> {
+        if self.message_size as usize > MAX_CMB_DATA_SIZE {
+            return Err(CaliptraError::RUNTIME_MAILBOX_API_REQUEST_DATA_LEN_TOO_LARGE);
+        }
+        let unused_byte_count = MAX_CMB_DATA_SIZE - self.message_size as usize;
+        Ok(&mut self.as_mut_bytes()[..size_of::<Self>() - unused_byte_count])
+    }
+}
+
+impl Request for CmEcdsaSignReq {
+    const ID: CommandId = CommandId::CM_ECDSA_SIGN;
+    type Resp = CmMldsaSignResp;
+}
+
+#[repr(C)]
+#[derive(Debug, IntoBytes, FromBytes, KnownLayout, Immutable, PartialEq, Eq)]
+pub struct CmEcdsaSignResp {
+    pub hdr: MailboxRespHeader,
+    pub signature_r: [u8; ECC384_SCALAR_BYTE_SIZE],
+    pub signature_s: [u8; ECC384_SCALAR_BYTE_SIZE],
+}
+
+impl Default for CmEcdsaSignResp {
+    fn default() -> Self {
+        Self {
+            hdr: MailboxRespHeader::default(),
+            signature_r: [0u8; ECC384_SCALAR_BYTE_SIZE],
+            signature_s: [0u8; ECC384_SCALAR_BYTE_SIZE],
+        }
+    }
+}
+
+impl Response for CmEcdsaSignResp {}
+
+// CM_ECDSA_VERIFY
+#[repr(C)]
+#[derive(Debug, IntoBytes, FromBytes, KnownLayout, Immutable, PartialEq, Eq)]
+pub struct CmEcdsaVerifyReq {
```

Comment:
Should we deprecate the other ECDSA verify command? They're very similar, but it probably makes sense to keep the CM one since it follows the semantics of the other CM commands.

Reply:
Hmm. We could, though the ECDSA384_SIGNATURE_VERIFY takes a public key hash, which people might be useful if the private key has not been loaded from the cryptographic mailbox or if the message is larger than the CM mailbox message limit.